pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -0,0 +1,789 @@
+		// resolved with deallocation.
+		podScheduling.Spec.SelectedNode = nodeName
+		logger.V(5).Info("start allocation", "pod", klog.KObj(pod), "node", klog.ObjectRef{Name: nodeName})
+		if err := state.publishPodScheduling(ctx, pl.clientset, podScheduling); err != nil {
+			return statusError(logger, err)
+		}
+		return statusUnschedulable(logger, "waiting for resource driver to allocate resource", "pod", klog.KObj(pod), "node", klog.ObjectRef{Name: nodeName})
+	}
+
+	// May have been modified earlier in PreScore or above.
+	if podSchedulingDirty {
+		if err := state.publishPodScheduling(ctx, pl.clientset, podScheduling); err != nil {
+			return statusError(logger, err)
+		}
+	}
+
+	// More than one pending claim and not enough information about all of them.
+	//
+	// TODO: can or should we ensure that scheduling gets aborted while
+	// waiting for resources *before* triggering delayed volume
+	// provisioning?  On the one hand, volume provisioning is currently
+	// irreversible, so it better should come last. On the other hand,
+	// triggering both in parallel might be faster.
+	return statusUnschedulable(logger, "waiting for resource driver to provide information", "pod", klog.KObj(pod))
+}
+
+// Unreserve clears the ReservedFor field for all claims.
+// It's idempotent, and does nothing if no state found for the given pod.
+func (pl *dynamicResources) Unreserve(ctx context.Context, cs *framework.CycleState, pod *v1.Pod, nodeName string) {
+	if !pl.enabled {
+		return
+	}
+	state, err := getStateData(cs)
+	if err != nil {
+		return
+	}
+	if len(state.claims) == 0 {
+		return
+	}
+
+	logger := klog.FromContext(ctx)
+	for index, claim := range state.claims {
+		if claim.Status.Allocation != nil &&
+			resourceclaim.IsReservedForPod(pod, claim) {
+			// Remove pod from ReservedFor.
+			claim := claim.DeepCopy()
+			reservedFor := make([]resourcev1alpha1.ResourceClaimConsumerReference, 0, len(claim.Status.ReservedFor)-1)
+			for _, reserved := range claim.Status.ReservedFor {
+				// TODO: can UID be assumed to be unique all resources or do we also need to compare Group/Version/Resource?
+				if reserved.UID != pod.UID {
+					reservedFor = append(reservedFor, reserved)
+				}
+			}
+			claim.Status.ReservedFor = reservedFor
+			logger.V(5).Info("unreserve", "resourceclaim", klog.KObj(claim))
+			if err := state.updateClaimStatus(ctx, pl.clientset, index, claim); err != nil {
+				// We will get here again when pod scheduling
+				// is retried.
+				logger.Error(err, "unreserve", "resourceclaim", klog.KObj(claim))
+			}
+		}
+	}
+}
+
+// PostBind is called after a pod is successfully bound to a node. Now we are
+// sure that a PodScheduling object, if it exists, is definitely not going to
+// be needed anymore and can delete it. This is a one-shot thing, there won't
+// be any retries.  This is okay because it should usually work and in those
+// cases where it doesn't, the garbage collector will eventually clean up.
```

Comment:
are you referring to the pod garbage collector deleting the owner pod?

Reply:
Yes.